scheduler/src/cook/mesos/api.clj

```diff
@@ -176,6 +177,15 @@
    (s/optional-key :extract) s/Bool
    (s/optional-key :cache) s/Bool})
 
+(def MesosComputeCluster
+  "Schema for a compute cluster"
+  {:framework-id s/Str})
+
+(def ComputeCluster
+  "Schema for a compute cluster"
+  {:compute-cluster-name s/Str
+   (s/optional-key :mesos) MesosComputeCluster})
```

Comment:
Should this also have `:type` with values like `:mesos` / `:kubernetes`?

Reply:
No need. Type is indicated based on the subdictionary key. If you have a :mesos subkey then it's a mesos type. If you have a :kubernetes subkey (yet to be done), you'll have a kubernetes type.

Reply:
I think the pattern we've used in other places is to have both the `type` and the sub-map

Reply:
Ok.